### DISCUSSION TITLE
User Studies/Surveys

### DISCUSSION THREAD

**Discussion by ProfDoof** [TriliumNext/Trilium#6076] [General]
Hey! So, before we get too deep into the weeds, I realized that I was thinking about the changes I would like to personally make to Trilium as a developer and realized that I don't really know how other people use Trilium, why they use it, and what they would want out of improvements to it. 

I think it would probably be a good idea to either have a survey constantly available for people to submit feedback/bug reports through that can be used to automatically create GitHub issues or discussions but also relieve the burden on users with less experience leveraging standard tech systems. 

I also think it would be a good idea to have a survey pop-up or something of that nature where users can say, hey, this is a thing I'm really missing, or other such feedback after they have been using Trilium for a while.

Another option depending on the size of the user base would be to have that survey pop up as multiple choice questions that they can submit or not, and those will give us insight into what the community is frustrated with.

Others probably know more than me about this as I am very new to thinking about such things, but I do think we need to be very careful to not get so excited about the possibility for new features we think are cool that we lose sight of the people that will keep this project alive through usage as well. 

Plus, who knows, seeing what people are actually having problems with may lead to other cool new features that other tools don't have.

**Comment by zerebos** [comment]
I think is a cool idea, and conceptually it would really be nice to get an understanding of how people use the application. But practically, I feel like there's no real way to reach a statistically significant number of current users and get the kind of information we need. If we post something on Trilium's current GitHub we'll get data biased likely towards developers that are active and looking at GitHub discussions. There are plenty of non-developer users of Trilium.

As for using a pop-up survey in current Trilium, we would need Zadam to do that which I'm not sure he would want to. Also from a UX standpoint, popups like that when you first open the app are often ignored and closed by users to their annoyance. I'm happy to be told I'm wrong but that's what I found during my research during grad school.

**Reply by ProfDoof** [reply]
I don't think it's something we would need to do now and something we should consider after refactoring. 

As for the pop-up thing, if we made it a notification type thing such that it is a negotiated interrupt where they are being made aware of an update is going out or something and we ask for feedback. But I'm not sure. I'd need to look into the research on the topic.

**Reply by meichthys** [reply]
I think if these types of studies/surveys were built-in but also opt-in, then there would be no problem with having the feedback be weighted toward those users who are willing to share. This way the project stays aligned with the users who are most invested in the project which is a healthy thing for an open source project with limited funding.

**Reply by Br0kenSilos** [reply]
I agree with @rauenzi about the GitHub data being biased to developer types. I'm in tech but I am not a developer personally. But my love of Trilium has "pulled me in" to GitHub. I know for a fact that I do not use Trilium to its fullest extent because some functions are over my head. Maybe some day..  Perhaps, simply have an easy to find "Report Issue" / "Submit feedback" button in the UX and then have it prompt the user based on what info they are wishing to submit.

**Comment by rmkni** [comment]
What about voting and discussing a backlog of features? We could see in time priorities of the majority.
Is it really an issue that the most invested people get to influence what's next for Trilium? I think it could if there are not enough people, so it should be easy to contribute.

**Reply by ProfDoof** [reply]
I'm not opposed to the idea of the most invested people driving development. 

However, I find it possible and maybe even likely that people who aren't as familiar with github and such platforms may want to use this and they are just as invested in the software but not in the development. I want to be sure that when we push the development forwards we don't accidentally "move their cheese" without some clear motivation and need. 

With that being said, one thought that I'm having after you said this is what if we just made a top level feature request/bug report page on trilium that allows them to be aware of things but then populate it from GitHub. We link to the GitHub issue or discussion, and add some links to documentation for how to create an account, how to create an issue, and how to vote for a feature request. 

Then we could just look at the things getting the most votes from there.

**Reply by Alumniminium** [reply]
> people who aren't as familiar with github and such

Don't forget that trilium is not your average note taking application. You can write frontend & backend scripts directly in notes. I think its fair to assume the average user has some programming background
